src/clusterfuzz/_internal/base/utils.py

```diff
@@ -124,6 +124,20 @@ def fetch_url(url):
   return response.text
 
 
+@retry.wrap(
+    retries=URL_REQUEST_RETRIES,
+    delay=URL_REQUEST_FAIL_WAIT,
+    function='base.utils.post_url')
+def post_url(url, data, headers):
```

Comment:
Let's document this method better:

- argument meanings, and ideally their types
- response meaning and type
- any errors it can raise
- the fact that it retries
- what timeout is applied